dockerfiles/cpubuilder_ubuntu_jammy_x86_64.Dockerfile

```diff
@@ -4,17 +4,25 @@ FROM ubuntu:jammy
 ######## Basic apt packages ########
 RUN apt update && \
     apt install -y \
-        wget python3.11-dev python3-pip git unzip curl gnupg2 lsb-release && \
-    update-alternatives --install /usr/local/bin/python python /usr/bin/python3.11 3 && \
-    update-alternatives --install /usr/local/bin/python3 python3 /usr/bin/python3.11 3 && \
+        wget git unzip curl gnupg2 lsb-release && \
     apt install -y \
         ccache clang-14 lld-14 libssl-dev ninja-build libxml2-dev llvm-dev pkg-config \
-        libcapstone-dev libtbb-dev libzstd-dev && \
-    apt clean && rm -rf /var/lib/apt/lists/*
+        libcapstone-dev libtbb-dev libzstd-dev
+# Python
+RUN apt install -y python3.11-dev python3.11-venv python3-pip && \
+    update-alternatives --install /usr/local/bin/python python /usr/bin/python3.11 3 && \
+    update-alternatives --install /usr/local/bin/python3 python3 /usr/bin/python3.11 3
+# Toolchains and build deps
+RUN apt install -y \
+        wget git unzip curl gnupg2 lsb-release \
+        ccache ninja-build clang-14 lld-14 gcc-9 g++-9 \
```

Comment:
gcc-11 was preinstalled on `ubuntu:jammy`. We could bump our min/tested version but this matches what IREE currently tests in `.github/workflows/ci_linux_x64_gcc.yml`.

Could also install a matrix of versions like what GitHub provides: https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2204-Readme.md#language-and-runtime